### ISSUE TITLE
Completions are not provided for action invocation

### ISSUE THREAD

**Issue by nadeeshaan** [ballerina-platform/ballerina-lang#29455]
**Description:**
Consider the following source snippet and $title at the cursor.

```
import ballerina/websocket as ws;

class FinnHubListener {
    private ws:Client wsClient;
    private decimal pollInterval;
    private boolean proceed = false;

    function init(ws:Client wsClient, decimal pollInterval) {
        self.wsClient = wsClient;
        self.pollInterval = pollInterval;
    }

    public function poll(string symbolName) {
        string subscriptionMsg = string `"type":"subscribe","symbol":"${symbolName}"}`;
        ws:Error? resp = self.wsClient->writeTextMessage(subscriptionMsg);
        if resp is ws:Error {
            return;
        }
        self.wsClient-><cursor>
        self.proceed = true;
        while self.proceed {

            io:println("Poll");
            runtime:sleep(self.pollInterval);
        }
    }

    public function stop() {
        io:println("interrupted!");
        self.proceed = false;
    }
}
```

**Affected Versions:**
sl-alpha3 at least

**Comment by nadeeshaan** [comment]
@pubudu91 If we try to get the type of the `self.wsClient` expression we either get CompilationErrorType or empty. Is there any way we can fix this? As I understood that is a limitation at the moment. If it so, we will have to revert back our LS implementation to depend on the custom symbol finder.

**Comment by pubudu91** [comment]
Isn't this expected, since the imports are missing? With the imports added, I'm still having an issue in getting completions for the above. In regular cases, the completion context for `v->` seems to be `AsyncSendActionNodeContext`. But in the above case, the completion context is `FunctionBodyBlockNodeContext`.

**Comment by nadeeshaan** [comment]
> Isn't this expected, since the imports are missing? With the imports added, I'm still having an issue in getting completions for the above. In regular cases, the completion context for `v->` seems to be `AsyncSendActionNodeContext`. But in the above case, the completion context is `FunctionBodyBlockNodeContext`.

Even when the import is there, the issue exists.

**Comment by nadeeshaan** [comment]
In the master, we can get the particular expression's type and the LS fix is added with #29455
